specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-04-01/registries.json

```diff
@@ -0,0 +1,911 @@
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/SkuTier"
+        }
+      },
+      "additionalProperties": false
+    },
+    "PartialUserAssignedIdentity": {
+      "type": "object",
+      "additionalProperties": false
+    },
+    "PrivateEndpointConnectionARMResource": {
+      "description": "Private endpoint connection definition.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "This is the private endpoint connection name created on SRP\r\nFull resource id: /subscriptions/{subId}/resourceGroups/{rgName}/providers/Microsoft.MachineLearningServices/{resourceType}/{resourceName}/privateEndpointConnections/{peConnectionName}",
+          "type": "string",
+          "x-nullable": true
+        },
+        "location": {
+          "description": "Same as workspace location.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "properties": {
+          "description": "Properties of the Private Endpoint Connection",
+          "$ref": "#/definitions/PrivateEndpointConnectionProperties",
+          "x-nullable": true
+        }
+      },
+      "additionalProperties": false
+    },
+    "PrivateEndpointConnectionProperties": {
+      "description": "Properties of the Private Endpoint Connection",
+      "type": "object",
+      "properties": {
+        "groupIds": {
+          "description": "The group ids",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "x-nullable": true
+        },
+        "privateEndpoint": {
+          "description": "The PE network resource that is linked to this PE connection.",
+          "$ref": "#/definitions/PrivateEndpointResource",
+          "x-nullable": true
+        },
+        "privateLinkServiceConnectionState": {
+          "description": "The connection state.",
+          "$ref": "#/definitions/PrivateLinkServiceConnectionState",
+          "x-nullable": true
+        },
+        "provisioningState": {
+          "description": "One of null, \"Succeeded\", \"Provisioning\", \"Failed\". While not approved, it's null.",
+          "type": "string",
+          "x-nullable": true
+        }
+      },
+      "additionalProperties": false
+    },
+    "PrivateEndpointResource": {
+      "description": "The PE network resource that is linked to this PE connection.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "e.g. /subscriptions/{networkSubscriptionId}/resourceGroups/{rgName}/providers/Microsoft.Network/privateEndpoints/{privateEndpointName}",
+          "type": "string",
+          "x-nullable": true
+        },
+        "subnetArmId": {
+          "description": "The subnetId that the private endpoint is connected to.",
+          "type": "string",
+          "x-nullable": true
+        }
+      },
+      "additionalProperties": false
+    },
+    "PrivateLinkServiceConnectionState": {
```

Comment:
>PrivateLinkServiceConnectionState [](http://example.com/codeflow?start=5&length=33)

common types same as before

Reply:
I see that in PrivateLinkServiceConnectionState the way it is defined in common-types, the status property includes only 3 states (pending, approved, rejected) but the way wee need it to be defined has a 4th state: disconnected, [EndpointServiceConnectionStatus.cs](https://msdata.visualstudio.com/Vienna/_git/vienna?path=/src/azureml-api/src/Common/PrivateLink.Contracts/Nrp/EndpointServiceConnectionStatus.cs&_a=contents&version=GBmaster)